source/postprocess/geoid.cc

```diff
@@ -0,0 +1,1025 @@
+                                  (this->get_geometry_model()).inner_radius();
+
+      // Get the value of the surface gravity acceleration from the gravity model
+      const Point<dim> surface_point = {outer_radius,0,0};
+      const double surface_gravity = this->get_gravity_model().gravity_vector(surface_point).norm();
+
+      // Get the value of the universal gravitational constant
+      const double G = aspect::constants::big_g;
+
+      // Get the spherical harmonic coefficients of the density contribution.
+      std::pair<std::vector<double>,std::vector<double> > SH_density_coes = density_contribution(outer_radius);
+
+      // Get the spherical harmonic coefficients of the surface and CMB dynamic topography
+      std::pair<std::pair<double, std::pair<std::vector<double>,std::vector<double> > >, std::pair<double, std::pair<std::vector<double>,std::vector<double> > > > SH_dyna_topo_coes;
+      SH_dyna_topo_coes = dynamic_topography_contribution(outer_radius,inner_radius);
+      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_surface_dyna_topo_coes = SH_dyna_topo_coes.first;
+      std::pair<double, std::pair<std::vector<double>,std::vector<double> > > SH_CMB_dyna_topo_coes = SH_dyna_topo_coes.second;
+
+      // Get the density contrast at the surface and CMB
+      const double surface_delta_rho =  SH_surface_dyna_topo_coes.first - density_above;
+      const double CMB_delta_rho = density_below - SH_CMB_dyna_topo_coes.first;
+
+      // Compute the spherical harmonic coefficients of geoid anomaly
+      std::vector<double> density_anomaly_contribution_coecos; // a vector to store cos terms of density anomaly contribution SH coefficients
+      std::vector<double> density_anomaly_contribution_coesin; // a vector to store sin terms of density anomaly contribution SH coefficients
+      std::vector<double> surface_dyna_topo_contribution_coecos; // a vector to store cos terms of surface dynamic topography contribution SH coefficients
+      std::vector<double> surface_dyna_topo_contribution_coesin; // a vector to store sin terms of surface dynamic topography contribution SH coefficients
+      std::vector<double> CMB_dyna_topo_contribution_coecos; // a vector to store cos terms of CMB dynamic topography contribution SH coefficients
+      std::vector<double> CMB_dyna_topo_contribution_coesin; // a vector to store sin terms of CMB dynamic topography contribution SH coefficients
+      std::vector<double> geoid_coecos;  // a vector to store cos terms of geoid anomaly SH coefficients
+      std::vector<double> geoid_coesin;  // a vector to store sin terms of geoid anomaly SH coefficients
+
+      // First compute the spherical harmonic contributions from density anomaly, surface dynamic topography and CMB dynamic topography
+      int ind = 0; // coefficients index
+      for (int ideg =  min_degree; ideg < max_degree+1; ideg++)
+        {
+          for (int iord = 0; iord < ideg+1; iord++)
```

Comment:
use `unsigned int` for the counter variables. Maybe you will need to convert it to a different type to use it in the formulas below.